src/gtest/test_mempool.cpp

```diff
@@ -136,3 +136,66 @@ TEST(Mempool, TxInputLimit) {
     EXPECT_EQ(state4.GetRejectReason(), "bad-txns-version-too-low");
 }
 
+
+TEST(Mempool, TxExpired) {
+    CTxMemPool pool(::minRelayTxFee);
+    bool missingInputs;
+
+    CMutableTransaction mtx;
+    mtx.fOverwintered = true;
+    mtx.nVersion = 3;
+    mtx.nFormatBranchId = 0xDEADBEEF;
+    // FIXME: Eventually replace with
+    // NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId;
+
+    // There is no active chain so expiry height is being compared to chainActive.Height() which is -1
+    // To simulate an expired tx, we must set an expiry height less than -1, that is, set a uint32 value
```

Comment:
This is confusing, there should never be a negative chain height. Can't we compare to a positive integer, or simulate a valid chain height? 

Also, why would an expired tx have a smaller nExpiryHeight than chainheight of `-1`? Should expire if it's >= to the chainheight.

Reply:
The issue is that when the unit test runs, there's no actual chain when calling into AcceptToMemoryPool, so chainActive.Height() returns -1.  I think a QA test will be better for testing expired txs as it will involve actually relaying txs from one node to another.  Setting tx expiry to -2 is a hack for the test, since it's less than -1, it will not be expired.  However, since the type at the moment is uint32_t and we need it to be int for the comparison against -1, we pick the uint32_t value which is the equivalent of -2 when it gets cast to an int.

Reply:
Ah ok, seems hacky, but I understand why this is necessary for the unit test now. Thanks for explaining.